### PR TITLE
Fixes #6659: Man pages for the rudder command

### PR DESCRIPTION
**Pull request by amousset**
http://www.rudder-project.org/redmine/issues/6659

### CODE REVIEW

**Comment by amousset** [comment]
First draft commit, using asciidoc and asciidoctor (for better html output). Still needs detailed doc for commands and a decent script to generate the source.

**Comment by amousset** [comment]
PR updated with an example of more detailed information.

**Comment by amousset** [comment]
PR updated with documentation.

File: README.md [hunk 1]
```diff
@@ -8,25 +8,39 @@ CLI
...
 - be called agent-&lt;command&gt; or server-&lt;command&gt;
 - parameters will not include the 'agent', 'server' or &lt;command&gt; given in the command line
 - the executable must contain a string with a single line of the form
-  # @description &lt;command description&gt;
+
+    # @description &lt;brief command description&gt;
+
+- the executable may also contain a single or multi-line string of the form
+
+    # @man &lt;detailed command description&gt;
```
Comment (peckpeck) [inline README.md]: If possible could you replace all those &lt and &gt with < and >

File: man/rudder.asciidoc.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+do
+echo "=== ${role}
+
+commands for rudder ${role}, run with *rudder* *${role}* _command_
+" >> ${OUTPUT}
+
+for file in `cd "${BASEDIR}"; find . -type f -name "${role}-*" | sort`
+   do
+   doc=`sed -ne "/#[ \t]*@description[ \t]*/s/#[ \t]*@description[ \t]*//p" "${BASEDIR}/${file}"`
+   man=`sed -ne "/#[ \t]*@man[ \t]*/s/#[ \t]*@man[ \t]*//p" "${BASEDIR}/${file}"`
```
Comment (peckpeck) [inline man/rudder.asciidoc.sh]: Very good idea

File: share/commands/agent-reinit [hunk 3]
```diff
@@ -1,5 +1,13 @@
 #!/bin/sh
 # @description re-initialise the agent to make it be seen as a new node on the server
+# @man This command will delete all local agent data, including its uuid and
+# @man keys, and also reset the agent internal state.
+# @man The only configuration kept is the server hostname or ip configured in _policy_server.dat_.
```
Comment (peckpeck) [inline share/commands/agent-reinit]: _policy_server.dat_. << will this be highlited properly ?

File: share/commands/agent-reinit [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
+# @man It will also send an inventory to the server, which will treat it as a new node inventory.
+# @man +
+# @man *WARNING*: This command will permanently delete your node uuid and keys, and no configuration will
+# @man applied before re-accepting and configuring the node on the server.
```
Comment (peckpeck) [inline share/commands/agent-reinit]: +be

File: share/commands/agent-reset [hunk 5]
```diff
@@ -1,5 +1,9 @@
 #!/bin/sh
 # @description reset agent status and cache
+# @man Remove all locks and state cache of the agent, and restore initial promises.
+# @man This won't affect the desired state of the node, but will only
+# @man reset the internal state of the agent, for example to test a rule
+# @man without caching interference.
```
Comment (peckpeck) [inline share/commands/agent-reset]: You can add a little sentence about using it when there is a problem with promise update.

File: share/commands/agent-update [hunk 6]
```diff
@@ -1,3 +1,5 @@
 #!/bin/sh
 # @description update promises on agent
-/var/rudder/cfengine-community/bin/cf-agent -KC -f failsafe.cf 
+# @man The agent will fetch its promises from its configured policy server.
```
Comment (peckpeck) [inline share/commands/agent-update]: + "the last version of" its promises

File: share/commands/server-debug [hunk 7]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
-# @description run a debug cf-serverd intended for a specific node
+# @description run a debug _cf-serverd_ intended for a specific node
+# @man This command targets a specific node and does not affect the running
+# @man infrastructure. It uses _iptables_ to redirect the specific node communications
+# @man to the port the debug server is listening on (5310 by default).
+
```
Comment (peckpeck) [inline share/commands/server-debug]: Add something about ctrl-c to stop

**Comment by peckpeck** [comment]
Thank you.
Please update according to comments.

File: share/commands/agent-reinit (hunk 3 above)
Comment (amousset) [inline share/commands/agent-reinit]: It will be underlined in the man page and in italics in the html output.

**Comment by amousset** [comment]
PR updated.

**Comment by peckpeck** [comment]
Good, Merging

**Pull request merged**